### PR TITLE
refactor: pass experiment arguments using a struct

### PR DESCRIPTION
**Pull request by bassosimone**
Closes https://github.com/ooni/probe/issues/2358.

### CODE REVIEW

File: internal/engine/experiment/dnscheck/dnscheck.go [hunk 1]
```diff
@@ -120,10 +120,11 @@ var (
 )
 
 // Run implements model.ExperimentSession.Run
-func (m *Measurer) Run(
-	ctx context.Context, sess model.ExperimentSession,
-	measurement *model.Measurement, callbacks model.ExperimentCallbacks,
-) error {
+func (m *Measurer) Run(ctx context.Context, args *model.ExperimentArgs) error {
+	_ = args.Callbacks
```
Comment (ainghazal) [inline internal/engine/experiment/dnscheck/dnscheck.go:124]: no need to assign here no?

Comment (bassosimone) [reply internal/engine/experiment/dnscheck/dnscheck.go:124]: No, I left it for clarity to highlight we're not using the callbacks. We may need to use them and the `_` will tell us so.

Comment (bassosimone) [reply internal/engine/experiment/dnscheck/dnscheck.go:124]: I mean: previously we were passing a `callbacks` argument and ignoring and I want to keep track of that.

**Comment by bassosimone** [comment]
I am going to merge having received a private "lgtm" from @ainghazal

**Pull request merged**